### PR TITLE
drivers: sensors: Convert to new DT_INST macros

### PR DESCRIPTION
**Pull request by galak**
Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: drivers/sensor/ak8975/ak8975.h [hunk 1]
```diff
@@ -23,13 +23,13 @@
 #define AK8975_MEASURE_TIME_US		9000
 #define AK8975_MICRO_GAUSS_PER_BIT	3000
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
-#if DT_INST_0_ASAHI_KASEI_AK8975_BASE_ADDRESS != 0x0C
+#if DT_INST_NODE_HAS_PROP(0, reg)
```
Comment (pabigot) [inline drivers/sensor/ak8975/ak8975.h]: This doesn't look right.  It seems to have lost something related to having support for an instance embedded in the mpu9150.

If that's intended I'd prefer to see this conversion pulled out to its own commit with an explanation, since the rest appear to be automated transformations.

**Review by pabigot — COMMENTED** [review]
Glanced at all, found one questionable change.  Ran tests on several devices on nrf52_pca20020.  Approval holding for  response to ak8975 question.

File: drivers/sensor/adxl362/adxl362.c [hunk 2]
```diff
@@ -1,4 +1,6 @@
 /* adxl362.c - ADXL362 Three-Axis Digital Accelerometers */
+
+#define DT_DRV_COMPAT adi_adxl362
```
Comment (MaureenHelm) [inline drivers/sensor/adxl362/adxl362.c]: move this below the license/copyright header

File: drivers/sensor/bme680/bme680.c [hunk 3]
```diff
@@ -4,6 +4,8 @@
  * https://www.bosch-sensortec.com/bst/products/all_products/bme680
  */
 
+#define DT_DRV_COMPAT bosch_bme680
```
Comment (MaureenHelm) [inline drivers/sensor/bme680/bme680.c]: Move below the copyright

File: drivers/sensor/bmm150/bmm150.c [hunk 4]
```diff
@@ -1,5 +1,7 @@
 /* bmm150.c - Driver for Bosch BMM150 Geomagnetic Sensor */
 
+#define DT_DRV_COMPAT bosch_bmm150
```
Comment (MaureenHelm) [inline drivers/sensor/bmm150/bmm150.c]: Move below the copyright

File: drivers/sensor/isl29035/isl29035.c [hunk 5]
```diff
@@ -1,5 +1,7 @@
 /* sensor_isl29035.c - driver for ISL29035 light sensor */
 
+#define DT_DRV_COMPAT isil_isl29035
```
Comment (MaureenHelm) [inline drivers/sensor/isl29035/isl29035.c]: Move below the copyright

File: drivers/sensor/isl29035/isl29035_trigger.c [hunk 6]
```diff
@@ -1,5 +1,7 @@
 /* sensor_isl29035.c - trigger support for ISL29035 light sensor */
 
+#define DT_DRV_COMPAT isil_isl29035
```
Comment (MaureenHelm) [inline drivers/sensor/isl29035/isl29035_trigger.c]: Move below the copyright

File: drivers/sensor/sx9500/sx9500.c [hunk 7]
```diff
@@ -1,5 +1,7 @@
 /* sensor_sx9500.c - Driver for Semtech SX9500 SAR proximity chip */
 
+#define DT_DRV_COMPAT semtech_sx9500
```
Comment (MaureenHelm) [inline drivers/sensor/sx9500/sx9500.c]: Move below the copyright

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/adxl362/adxl362.c (hunk 2 above)
Comment (galak) [reply drivers/sensor/adxl362/adxl362.c]: done

File: drivers/sensor/sx9500/sx9500.c (hunk 7 above)
Comment (galak) [reply drivers/sensor/sx9500/sx9500.c]: done

File: drivers/sensor/isl29035/isl29035_trigger.c (hunk 6 above)
Comment (galak) [reply drivers/sensor/isl29035/isl29035_trigger.c]: done

File: drivers/sensor/isl29035/isl29035.c (hunk 5 above)
Comment (galak) [reply drivers/sensor/isl29035/isl29035.c]: done

File: drivers/sensor/bmm150/bmm150.c (hunk 4 above)
Comment (galak) [reply drivers/sensor/bmm150/bmm150.c]: done

File: drivers/sensor/bme680/bme680.c (hunk 3 above)
Comment (galak) [reply drivers/sensor/bme680/bme680.c]: done

File: drivers/sensor/ak8975/ak8975.h (hunk 1 above)
Comment (galak) [reply drivers/sensor/ak8975/ak8975.h]: It wasn't intended, just a case the script didn't handle correctly.  Fixing it up.

File: drivers/sensor/ak8975/ak8975.h [hunk 8]
```diff
@@ -23,13 +23,13 @@
...
-#ifdef DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
-#if DT_INST_0_ASAHI_KASEI_AK8975_BASE_ADDRESS != 0x0C
+#if DT_NODE_HAS_PROP(DT_INST(0, invensense_mpu9150), reg)
+#if DT_REG_ADDR(DT_INST(0, asahi_kasei_ak8975)) != 0x0C
 #error "I2C address must be 0x0C when AK8975 is part of a MPU9150 chip"
 #endif
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150
-#define MPU9150_I2C_ADDR		DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
+#if DT_HAS_DRV_INST(0)
```
Comment (pabigot) [inline drivers/sensor/ak8975/ak8975.h]: This still seems wrong, or at least confusing.  `DRV_COMPAT` here is `asahi_kasei_ak8975` not `invensense_mpu9150`, so the change suggests it's looking at a different node now.

I think this should be:

```
#if DT_NODE_HAS_PROP(DT_INST(0, invensense_mpu9150), reg)
#if DT_REG_ADDR(DT_INST(0, asahi_kasei_ak8975)) != 0x0C
#error "I2C address must be 0x0C when AK8975 is part of a MPU9150 chip"
#endif

#define MPU9150_I2C_ADDR                DT_INST_REG_ADDR(0)

#define MPU9150_REG_BYPASS_CFG          0x37
...
```
i.e. remove the condition around defining `MPU9150_I2C_ADDR`.  It was never necessary anyway: the context requires that `invensense_mpu9150`  exist with a `reg` property and that instance 0 of asahi_kasei_ak8975 have a `reg` property with a matching value, so you can just use it.

**Review by pabigot — CHANGES_REQUESTED** [review]

File: drivers/sensor/ak8975/ak8975.h (hunk 8 above)
Comment (galak) [reply drivers/sensor/ak8975/ak8975.h]: Ok, should be fixed up now.

File: drivers/sensor/ak8975/ak8975.h [hunk 9]
```diff
@@ -23,22 +23,20 @@
...
-#if DT_INST_0_ASAHI_KASEI_AK8975_BASE_ADDRESS != 0x0C
+#if DT_NODE_HAS_PROP(DT_INST(0, invensense_mpu9150), reg)
+#if DT_REG_ADDR(DT_INST(0, asahi_kasei_ak8975)) != 0x0C
 #error "I2C address must be 0x0C when AK8975 is part of a MPU9150 chip"
 #endif
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150
-#define MPU9150_I2C_ADDR		DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
-#endif
+#define MPU9150_I2C_ADDR		DT_REG_ADDR(DT_INST(0, invensense_mpu9150))
```
Comment (pabigot) [inline drivers/sensor/ak8975/ak8975.h]: I was expecting `DT_INST_REG_ADDR(0)` (two places, if we're following driver idiom for this header), and this isn't going to be right if the instance ordinal for `invensense,mpu9150` isn't the same as the instance ordinal for `asahi-kasei,ak8975` (#20287) but I guess this is generally equivalent.

**Review by pabigot — APPROVED** [review]

**Review by franciscomunoz — APPROVED** [review]

File: drivers/sensor/bme680/bme680.c (hunk 3 above)
Comment (MaureenHelm) [reply drivers/sensor/bme680/bme680.c:7]: this one still needs to be done

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please don't rebase when you fix this. That way I can just review the force push diff instead of the whole patch again.

**Comment by galak** [comment]
> Please don't rebase when you fix this. That way I can just review the force push diff instead of the whole patch again.

sorry about that.

**Comment by galak** [comment]
Fixed bme680.c

**Review by MaureenHelm — APPROVED** [review]
